app/fight.py

```diff
@@ -0,0 +1,22 @@
+from app.knights import Knight
+
+
+class Arena:
+    @staticmethod
+    def opposition(knights: dict) -> dict:
+        dict_knights = {}
+        for key, value in knights.items():
+            knight = Knight(value["name"], value["power"],
+                            value["hp"])
+            knight.use_weapon(value["weapon"]["power"])
+            if value["armour"] is not None:
```

Comment:
I recommend to read about `:=` in Python, cool operator, but it is not necessary